service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/CafcassEmailContentProvider.java

```diff
@@ -28,7 +37,17 @@ public SubmitCaseCafcassTemplate buildCafcassSubmissionNotification(CaseDetails
 
         template.setCafcass(cafcassLookupConfiguration.getCafcass(localAuthorityCode).getName());
         template.setLocalAuthority(localAuthorityNameLookupConfiguration.getLocalAuthorityName(localAuthorityCode));
+        template.setDocumentLink(linkToAttachedDocument(caseData.getSubmittedForm()));
 
         return template;
     }
+
+    private Map<String, Object> linkToAttachedDocument(final DocumentReference documentReference) {
+        return Optional.ofNullable(documentReference)
+            .map(DocumentReference::getBinaryUrl)
+            .map(documentDownloadService::downloadDocument)
+            .flatMap(NotifyAttachedDocumentLinkHelper::generateAttachedDocumentLink)
+            .map(JSONObject::toMap)
+            .orElseThrow(() -> new DocumentException());
+    }
```

Comment:
Shared between both CafcassEmailContentProvider and HmctsEmailContentProvider. Could we extract this? Means we can test it also.

Reply:
Have extracted to AbstractEmailContentProvider, no test added as agreed that it is sufficiently tested by its method calls